### PR TITLE
Payments: Update `getPayments` call to limit the number of payments

### PR DESCRIPTION
**Pull request by shubhamkmr04**
Issue: #2605

### CODE REVIEW

File: backends/LightningNodeConnect.ts [hunk 1]
```diff
@@ -170,10 +170,11 @@ export default class LightningNodeConnect {
                 route_hints: data.route_hints
             })
             .then((data: lnrpc.AddInvoiceResponse) => snakeize(data));
-    getPayments = async () =>
+    getPayments = async (data: any) =>
         await this.lnc.lnd.lightning
             .listPayments({
-                include_incomplete: true
+                include_incomplete: true,
+                max_payment: data && data?.maxPayments
```
Comment (kaloudis) [inline backends/LightningNodeConnect.ts:177]: `max_payments` plural?

Comment (shubhamkmr04) [reply backends/LightningNodeConnect.ts:177]: tested it on only LND yet.
making it a draft PR. gonna work on other interfaces as well

File: stores/PaymentsStore.ts [hunk 2]
```diff
@@ -30,10 +30,16 @@ export default class PaymentsStore {
     };
 
     @action
-    public getPayments = async () => {
+    public getPayments = async (
+        maxPayments: any = undefined,
```
Comment (kaloudis) [inline stores/PaymentsStore.ts:34]: I believe this should be of type `number`

File: stores/PaymentsStore.ts [hunk 3]
```diff
@@ -30,10 +30,13 @@ export default class PaymentsStore {
     };
 
     @action
-    public getPayments = async () => {
+    public getPayments = async (maxPayments?: number, reversed?: boolean) => {
```
Comment (kaloudis) [inline stores/PaymentsStore.ts:33]: let's pass in the params as a single object so it's easier to pass in just one or the other

**Review by kaloudis — CHANGES_REQUESTED** [review]

**Comment by kaloudis** [comment]
@shubhamkmr04 did you want to make any more changes here or is this ready for testing and review?

**Comment by shubhamkmr04** [comment]
> @shubhamkmr04 did you want to make any more changes here or is this ready for testing and review?

Its good to go i think

**Review by kaloudis — CHANGES_REQUESTED** [review]
Tested. Everything looks good but we need to change `backends/EmbeddedLND` as well

**Review by kaloudis — APPROVED** [review]
tACK

**Pull request merged**